sys/net/application_layer/dns/dns.c

```diff
@@ -129,6 +131,13 @@ static int _parse_dns_reply(uint8_t *buf, size_t len, void* addr_out, int family
             bufpos += addrlen;
             continue;
         }
+        if (addrlen > SOCK_DNS_MAX_ADDR_LEN) {
+            return -EINVAL;
```

Comment:
Shouldn't this be `EBADMSG`?

Reply:
I was thinking, since this isn't a bad message per se, but only invalid within the restrictions of this application, that this should be a different error class.

Reply:
But since this is supplied to the user and `EINVAL` usually means invalid parameter, a different value might be better.

Reply:
I take `ERANGE` instead.

Reply:
This code is only reached for `IN/A` and `IN/AAAA`, whose lengths must be exactly 4 or 16 bytes, as mandated by RFC1035 and RFC3596, respectivly.

Reply:
Fixed.